gcc/config/riscv/constraints.md

```diff
@@ -168,6 +171,23 @@
        (and (match_test "IN_RANGE (ival, 0, 1073741823)")
             (match_test "exact_log2 (ival + 1) != -1"))))
 
+(define_constraint "CVmp"
+  "An address for post-modify or reg+reg stores and loads"
+  (and (match_code "mem")
+       (match_test "GET_CODE (XEXP (op, 0)) == POST_MODIFY
+		|| (GET_CODE (XEXP (op, 0)) == PLUS
+		&& GET_CODE (XEXP (XEXP (op, 0), 0)) == REG
+                && GET_CODE (XEXP (XEXP (op, 0), 1)) == REG)"))
+)
+
+(define_constraint "CVmr"
+  "An address for reg+reg stores and loads"
+  (and (match_code "mem")
```

Comment:
Again, instead of `match_code "mem"`, reuse the default mem constraint `(match_operand 0 "TARGET_MEM_CONSTRAINT")`.